File: client/web/sudoo/lib/domain/model/auth/role.dart
enum Role {
  STAFF("STAFF"),
  ADMIN("ADMIN"),
  CONSUMER("CONSUMER");

  final String value;

  const Role(this.value);

  factory Role.fromValue(String value) {
    switch(value) {
      case "STAFF":
        return Role.STAFF;
      case "ADMIN":
        return Role.ADMIN;
      default:
        return Role.CONSUMER;
    }
  }
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/auth/Role.kt
package com.sudo248.sudoo.domain.entity.auth

enum class Role(val value: String) {
    CONSUMER("Thành viên"),
    STAFF("Nhân viên");

    companion object {
        fun fromStringValue(value: String): Role {
            if (value == "Nhân viên") return STAFF
            return CONSUMER
        }
    }
}